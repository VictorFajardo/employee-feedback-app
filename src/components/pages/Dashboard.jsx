// React components
import { useEffect } from 'react';
import { Outlet } from 'react-router-dom';
// Redux components
import { useAppSelector, useAppDispatch } from "../../app/hooks";
import { currentUser } from '../../features/authentication/authenticationSlice';
import { setEmployees } from '../../features/employees/employeesSlice';
import { setReviews } from '../../features/reviews/reviewsSlice';
// Api components
import { getUsersApi, getReviewsApi } from '../../utilities/firebase';
// Material components
import Header from './header/header';
import Menu from './menu/menu';
import Divider from '@mui/material/Divider'

const Dashboard = () => {
  const dispatch = useAppDispatch();
  const user = useAppSelector(currentUser); // Select the current user to verify level access

  useEffect(() => {
    if (!user || !user.admin) return;

    const getUsersMap = async () => {
      // Api to retrive the users from the users collection
      const usersMap = await getUsersApi();
      // Reducer to set users into the state manager
      dispatch(setEmployees(usersMap));
    };
    
    return getUsersMap;
  }, [dispatch, user]);

  useEffect(() => {
    if (!user) return;

    const getReviewsMap = async () => {
      // Api to retrive the users from the users collection
      const reviewsMap = await getReviewsApi();
      // Reducer to set users into the state manager
      if (user.admin) dispatch(setReviews(reviewsMap)); // Admin: All the reviews
      else dispatch(setReviews(reviewsMap.filter((review) => user.reviews.includes(review.id)))); // User: Only his/her assigned reviews
    };

    return getReviewsMap;
  }, [dispatch, user]);

  return (
    <>
      <Header />
      <Menu />
      <Divider variant="middle" />
      <Outlet />
    </>
  )
}

export default Dashboard
