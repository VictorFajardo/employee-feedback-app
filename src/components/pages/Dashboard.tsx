// React components
import { useEffect } from 'react';
import { Outlet } from 'react-router-dom';
// Redux components
import { useAppSelector, useAppDispatch } from "../../app/hooks";
import { currentUser } from '../../features/authentication/authenticationSlice';
import { setEmployees } from '../../features/employees/employeesSlice';
import { setReviews } from '../../features/reviews/reviewsSlice';
// Api components
import { getUsersApi, getReviewsApi } from '../../utilities/firebase';
// Material components
import Header from './header/header';
import Menu from './menu/menu';
import Divider from '@mui/material/Divider'

const Dashboard = () => {
  const dispatch = useAppDispatch();
  const user = useAppSelector(currentUser); // Select the current user to verify level access

  useEffect(() => {
    if (!user || !user.admin) return;

    async function getUsersMap (){
      // Api to retrive the users from the users collection
      const usersMap = await getUsersApi();
      // Reducer to set users into the state manager
      dispatch(setEmployees(usersMap));
    };
    
    getUsersMap();
  }, [dispatch, user]);

  useEffect(() => {
    if (!user) return;

    async function getReviewsMap() {
      // Api to retrive the users from the users collection
      const reviewsMap = await getReviewsApi();
      // Reducer to set users into the state manager
      if (user?.admin) dispatch(setReviews(reviewsMap)); // Admin: All the reviews
      else dispatch(setReviews(reviewsMap.filter((review) => user?.reviews.includes(review.id)))); // User: Only his/her assigned reviews
    };

    getReviewsMap();
  }, [dispatch, user]);

  return (
    <>
      <Header />
      <Menu />
      <Divider variant="middle" />
      <Outlet />
    </>
  )
}

export default Dashboard
