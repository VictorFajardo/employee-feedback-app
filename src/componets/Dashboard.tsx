// React components
import { useEffect } from 'react';
import { Outlet } from 'react-router-dom';
// Redux components
import { useAppSelector, useAppDispatch } from '../hooks';
import { currentUser } from '../features/authentication/authenticationSlice';
import { setEmployees } from '../features/employees/employeesSlice';
import { setReviews } from '../features/reviews/reviewsSlice';
// Api components
import { getUsersApi, getReviewsApi } from '../utilities/firebase';
// Material components

import Header from './header/header';
import Menu from './menu/menu';
import Divider from '@mui/material/Divider';

const Dashboard = () => {
  const dispatch = useAppDispatch();
  const user = useAppSelector(currentUser); // Select the current user to verify level access

  useEffect(() => {
    if (!user || !user.admin) return;

    async function getUsersMap() {
      // Api to retrive the users from the users collection
      const usersMap = await getUsersApi();
      // Reducer to set users into the state manager
      dispatch(setEmployees(usersMap));
    }

    getUsersMap();
  }, [dispatch, user]);

  useEffect(() => {
    if (!user) return;

    async function getReviewsMap() {
      // Api to retrive the users from the users collection
      const reviewsMap = await getReviewsApi(user!.admin!, user!.email!);
      // Reducer to set users into the state manager
      dispatch(setReviews(reviewsMap));
    }

    getReviewsMap();
  }, [dispatch, user]);

  return (
    <>
      <Header />
      <Menu />
      <Divider variant="middle" />
      <Outlet />
    </>
  );
};

export default Dashboard;
