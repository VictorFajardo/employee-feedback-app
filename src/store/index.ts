import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import employeesSlice from '../features/employees/employeesSlice';
import reviewsSlice from '../features/reviews/reviewsSlice';
import authenticationSlice from '../features/authentication/authenticationSlice';

export const store = configureStore({
  reducer: {
    employees: employeesSlice,
    reviews: reviewsSlice,
    authentication: authenticationSlice,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
