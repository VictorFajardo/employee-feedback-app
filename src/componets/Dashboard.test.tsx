import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { store } from '../store';
import Dashboard from './Dashboard';

it('expect render Dashboard component', () => {
  expect(
    shallow(
      <Provider store={store}>
        <Dashboard />
      </Provider>
    )
  ).toMatchSnapshot();
});
