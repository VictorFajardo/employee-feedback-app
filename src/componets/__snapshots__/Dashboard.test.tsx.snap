// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect render Dashboard component 1`] = `ShallowWrapper {}`;
